### PR TITLE
do not include requiredJvmOpts in launcher-check.yml

### LINKED ISSUE #152: rotate GC logs?

### PR DESCRIPTION
**Pull request by gracew**
fixes https://github.com/palantir/gradle-java-distribution/issues/152

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/gradle-java-distribution/156)

### CODE REVIEW

**Comment by gracew** [comment]
@markelliot @uschi2000 can you review?

File: src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy [hunk 1]
```diff
@@ -309,7 +309,6 @@ class JavaDistributionPluginTests extends GradleTestSpec {
 
         then:
         String startScript = file('dist/service-name-0.1/service/bin/service-name', projectDir).text
-        startScript.contains('DEFAULT_JVM_OPTS=\'"-Djava.io.tmpdir=var/data/tmp" "-XX:+PrintGCDateStamps" "-XX:+PrintGCDetails" "-XX:-TraceClassUnloading" "-XX:+UseGCLogFileRotation" "-XX:GCLogFileSize=10M" "-XX:NumberOfGCLogFiles=10" "-Xloggc:var/log/gc-%t-%p.log" "-verbose:gc" "-Xmx4M" "-Djavax.net.ssl.trustStore=truststore.jks"\'')
```
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy:312]: why does this go away?

Comment (gracew) [reply src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy:312]: oops, this should still include the last two options...

**Comment by uschi2000** [comment]
Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy, line 363 at r2](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156#-KaghuX66LmRlz1lwhyI:-KaghuX66LmRlz1lwhyJ:b-c30i2k) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/940e7dc4599bf67c6c87f9a8b108b32bb2c8a01b/src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy#L363)):*
> ```Groovy
>         def expectedCheckConfig = expectedStaticConfig
>         expectedCheckConfig.setJvmOpts([
>                 '-Xmx4M',
> ```

should include tmpdir config, i think?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156)*

**Comment by gracew** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy, line 363 at r2](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156#-KaghuX66LmRlz1lwhyI:-KahDQDaJKNweHs_vvV-:b-lvnvm1) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/940e7dc4599bf67c6c87f9a8b108b32bb2c8a01b/src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy#L363)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

should include tmpdir config, i think?
</blockquote></details>

nope. this is for the check config, so I only pass through the user-specified java opts.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156)*

**Comment by gracew** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy, line 363 at r2](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156#-KaghuX66LmRlz1lwhyI:-KahDitwhrZeRmX-IXy5:b-a9y70g) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/940e7dc4599bf67c6c87f9a8b108b32bb2c8a01b/src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy#L363)):*
<details><summary><i>Previously, gracew (Grace Wang) wrote…</i></summary><blockquote>

nope. this is for the check config, so I only pass through the user-specified java opts.
</blockquote></details>

if you prefer I can separate the tmpdir and the GC options into two different variables so that tmpdir is always specified but GC is excluded from the check config.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156)*

**Comment by uschi2000** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy, line 312 at r3](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156#-KalO_irIacals62Cqfy:-KalO_irIacals62Cqfz:bm6f7r) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/bc9700c63b8efa0aceb5650487ccb598a57c2265/src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy#L312)):*
> ```Groovy
>         then:
>         String startScript = file('dist/service-name-0.1/service/bin/service-name', projectDir).text
>         startScript.contains('DEFAULT_JVM_OPTS=\'"-Xmx4M" "-Djavax.net.ssl.trustStore=truststore.jks"\'')
> ```

this is stale. i'll make a follow-up PR to fix it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156)*

**Review by uschi2000 — APPROVED** [review]

**Comment by uschi2000** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy, line 312 at r3](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156#-KalO_irIacals62Cqfy:-KalOwgfRomdfxrKNI8c:bb6ywmx) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/bc9700c63b8efa0aceb5650487ccb598a57c2265/src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy#L312)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

this is stale. i'll make a follow-up PR to fix it.
</blockquote></details>

LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156)*

**Comment by uschi2000** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy, line 312 at r3](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156#-KalO_irIacals62Cqfy:-KalQN6-xrHG8K1iZczC:b2ni8w0) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/bc9700c63b8efa0aceb5650487ccb598a57c2265/src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy#L312)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

LGTM
</blockquote></details>

nvm

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/156)*

**Pull request merged**